Allow limiting calendar field to current year

ggppdk opened the pull request:
Pull Request for Issue #19831

@dgt41 , your opinion, i think this change matches documentation

### Summary of Changes
Minimum and Maximum (relative) year setting is not to added by the layout when its value is '0'
Change it so that it is, thus you can get
data-min-year="0"
data-max-year="0"

### Testing Instructions
Create a calendar field in XML that has attributes
minyear="0"
maxyear="0"


### Expected result
The selectable year in the calender is limited to current year


### Actual result
No limit


### Documentation Changes Required
None

ggppdk committed:
Allow locking to min and / or max year to current year

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on e216889cad72949e16b1c4bee384fb26f929d0b3<br /><br />using the xml in the original issue to test with I find no change after this pr<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>

ggppdk committed:
Update calendar.php

ggppdk commented:
@brianteeman 

It should be good now,

1st commit of this PR was only enough for JHtml::calendar , which currently completely ignores minYear, maxYear attributes
I will make a different PR for JHtml::calendar

ggppdk commented:
Here is PR to add support for minYear, maxYear attributes to JHtml::calendar too
https://github.com/joomla/joomla-cms/pull/19847

brianteeman commented:
wondering why the layout needed the change. JUST using the library change seems enough

ggppdk commented:
The layout uses code 
empty(...) which cannot handle string value '0', without this change value '0' cannot be added

brianteeman commented:
Sorry - forgot i was also testing the -1 etc at that time<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>

brianteeman commented:
I have tested this item :white_check_mark: successfully on f27cbd61848858f866cef472aafdfd28aaa8c3b2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>

brianteeman commented:
The unit tests will need updating

1) JHtmlTest::testCalendar
Undefined variable: minYear
/home/travis/build/joomla/joomla-cms/tests/unit/core/helper.php:55
/home/travis/build/joomla/joomla-cms/layouts/joomla/form/field/calendar.php:128
/home/travis/build/joomla/joomla-cms/libraries/src/Layout/FileLayout.php:125
/home/travis/build/joomla/joomla-cms/libraries/src/Layout/LayoutHelper.php:73
/home/travis/build/joomla/joomla-cms/libraries/src/HTML/HTMLHelper.php:1086
/home/travis/build/joomla/joomla-cms/tests/unit/suites/libraries/cms/html/JHtmlTest.php:1449
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>

ggppdk committed:
Cleared non-set variable notices

ggppdk commented:
> The unit tests will need updating

I missed this, fixed it

The unit tests are still correct, just they caught the non-set variable PHP notices when the minyear and maxyear are not given for the calendar field, which is what unit tests test by default

added check for non set variables $minYear, $maxYear,
behaviour of PR was not changed by this last commit

dgrammatiko commented:
I have tested this item :white_check_mark: successfully on 243a1af670ebdae036470843d2c9a9e919bc46bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19846">issues.joomla.org/tracker/joomla-cms/19846</a>.</sub>
